tests/gis_tests_/tests.py

```diff
@@ -8,6 +8,51 @@
 @skipUnlessDBFeature("gis_enabled")
 class LookupTests(TestCase):
     def test_unsupported_lookups(self):
-        msg = "MongoDB does not support the same_as lookup."
+        msg = "MongoDB does not support the 'same_as' lookup."
         with self.assertRaisesMessage(NotSupportedError, msg):
             City.objects.get(point__same_as=Point(95, 30))
+
+    def test_within_lookup(self):
+        city = City.objects.create(point=Point(95, 30))
+        qs = City.objects.filter(point__within=Point(95, 30).buffer(10))
+        self.assertIn(city, qs)
```

Comment:
It would be nice to have more than one test geometry to show that the operator does some filtering. `assertIn()` isn't the best assertion since the queryset could inadvertently return all objects without any filtering.